coral/src/app/features/components/table-filters/OperationTypeFilter.tsx

```diff
@@ -0,0 +1,58 @@
+import { NativeSelect } from "@aivenio/aquarium";
+import { useSearchParams } from "react-router-dom";
+import {
+  requestOperationTypeNameMap,
+  operationTypeList,
+} from "src/app/features/approvals/utils/request-operation-type-helper";
+import { RequestOperationType } from "src/domain/requests/requests-types";
+import { ResolveIntersectionTypes } from "types/utils";
+
+type RequestOperationTypeOptions = ResolveIntersectionTypes<
+  RequestOperationType | "ALL"
+>;
+
+function OperationTypeFilter() {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const operationType =
+    (searchParams.get("operationType") as RequestOperationTypeOptions | null) ??
+    "ALL";
+
+  const handleChangeOperationType = (
+    nextOperationType: RequestOperationTypeOptions
+  ) => {
+    if (nextOperationType === "ALL") {
+      searchParams.delete("operationType");
+      searchParams.set("page", "1");
+      setSearchParams(searchParams);
+    } else {
+      searchParams.set("operationType", nextOperationType);
+      searchParams.set("page", "1");
+      setSearchParams(searchParams);
+    }
+  };
+
+  return (
+    <NativeSelect
+      labelText={"Filter by operation type"}
+      key={"filter-operationType"}
+      defaultValue={operationType}
+      onChange={(e) => {
+        const operationType = e.target.value as RequestOperationTypeOptions;
+        return handleChangeOperationType(operationType);
+      }}
+    >
+      <option key={"ALL"} value={"ALL"}>
+        All operation types
+      </option>
+      {operationTypeList.map((operationType) => {
+        return (
+          <option key={operationType} value={operationType}>
+            {requestOperationTypeNameMap[operationType]}
+          </option>
+        );
+      })}
+    </NativeSelect>
+  );
+}
+
+export default OperationTypeFilter;
```

Comment:
We started out with only doing named exports (`export { OperationTypeFilter }`), mainly because it makes (automated) refactoring easier. I think we should try to stick to that pattern so it's consistent 🤔  (it's not super important for this PR right now, we could do one later to update all & also introduce a linter rule)

Reply:
Ah, that's nice. I also prefer named exports :o

Reply:
[cce9778](https://github.com/aiven/klaw/pull/888/commits/cce9778c39177cb69493424a4abb9ea240341a48)